pages/common/nix-shell.md

```diff
@@ -0,0 +1,28 @@
+# nix-shell
+
+> Start an interactive shell based on a Nix expression.
+> More information: <https://nixos.org/manual/nix/stable/#sec-nix-shell>.
+
+- Start with nix expression in `shell.nix` or `default.nix` in the current directory:
+
+`nix-shell`
+
+- Run shell command in non-interactive shell and exit:
+
+`nix-shell --run "{{command}} {{command_arguments}}"`
+
+- Start with expression in `default.nix` in the current directory:
+
+`nix-shell {{default.nix}}`
+
+- Start with packages loaded from nixpkgs:
+
+`nix-shell --packages {{package_name_1}} {{package_name_2}}`
+
+- Start with packages loaded from specific nixpkgs revision:
+
+`nix-shell --packages {{package_names}} -I nixpkgs={{https://github.com/NixOS/nixpkgs/archive/{{nixpkgs_revision_hash}}.tar.gz}}`
```

Comment:
Oh, we have nested `{{`:
```suggestion
`nix-shell --packages {{package_names}} -I nixpkgs={{https://github.com/NixOS/nixpkgs/archive/nixpkgs_revision_hash.tar.gz}}`
```

Reply:
Could we use something more generic? Or the packages only comes from the `nixpkgs` repository?
```suggestion
`nix-shell --packages {{package_names}} -I nixpkgs={{https://example.com/nix_package.tar.gz}}`
```

Reply:
I think we could, and they could come from a tarball with a single top-level directory or a path to a local file. I'm inclined to say that this example is most valuable if it references `nixpkgs` directly, because that seems like the more common use-case by far, and if you are doing something else, you probably already have a more advanced understanding of nix / this option.

Reply:
I updated the placeholder to be `nixpkgs_revision` instead of `nixpkgs_revision_hash` to not rule out other references, let me know what you think?